build_tools/waf_dynamo.py

```diff
@@ -1649,7 +1642,8 @@ def detect(conf):
         conf.env['AR']       = '%s/%s-ar' % (bintools, tool_name)
         conf.env['RANLIB']   = '%s/%s-ranlib' % (bintools, tool_name)
         conf.env['LD']       = '%s/lld' % (bintools)
-        conf.env['DX']       = '%s/android-sdk/build-tools/%s/dx' % (ANDROID_ROOT, ANDROID_BUILD_TOOLS_VERSION)
+
+        conf.find_program('d8', var='D8', mandatory = True, path_list=[sdkinfo['build_tools']])
```

Comment:
Making it a "find_program" makes sure we get a nice error if the `d8` (and in effect, the sdk) is missing.